### ISSUE TITLE
Something weird with the import paths 

### ISSUE THREAD

**Issue by snixer724** [Seanba/SuperTiled2Unity#221]
Hey everyone, we've been using ST2U and Tiled for like 4-5 years now and just recently upgraded from Unity 2018.3f to 2022.1.9. We also upgraded ST2U from 1.9.1 to 1.10.5 and Tiled from 1.8 to 1.9. 

This project has been working fine the entire time until the upgrade. We are missing each map prefab. 

I've attached a copy of the thread in Tiled chat. 
![image](https://user-images.githubusercontent.com/3958802/179403372-daeaba80-d38b-4c3f-8e9f-d12607f9702c.png)

The problem is the missing asset piece, where the TSX cannot find the PNG. If you see the thread you will find that all of the paths look correct. And the weird thing is that randomly I'll click the "Reimport" button on unity and sometimes it doesn't work, and sometimes it does magically stop having an error. With no change at all.

If I can consistently figure out what is allowing the connection to be made I will update here, but I was wondering if anyone here might have some ideas? 

What I was hoping to do was just do a reimport from ST2U and it would fix all of them but that hasn't worked. Things I've tried are.

1) Reimport from ST2U unity option
2) Manually change TSX file to point somewhere wrong and then correct it (this is what seems to potentially work sometimes?)
3) Delete all meta data to regenerate. 
4) Smash the reimport button on unity.
5) Downgrade ST2U to 1.9.3 / 1.10.4 / 1.10.3 and tried. No luck.
6) Use Tiled 1.8 compatiability. No luck.

**Comment by snixer724** [comment]
I've read issue 67 and the duplicate it closes as well. I don't think that is the issue here.

**Comment by Seanba** [comment]
I'll take a look when I get your project but I wonder if a jump from ST2U `1.9.1` to `1.10.5` is the culprit. I would try deleting the ST2U folder entirely and then reinstall the ST2U Unity package.

This would be a good argument to finally make ST2U distributed through the Package Manager.

**Comment by snixer724** [comment]
Issue is not there in Unity 2021, but it is in 2022. Looks like something in 2022 is not compatiable.

**Comment by erikedling** [comment]
I got the exact same problem after I upgraded unity as well. I found a way to at least make it work until there is a fix(but extremely time consuming).

1. First I might get an error when trying to import a map saying that the tileset must be imported, for example Assets/Maps/TileSets/Act3_Inside_House_Items.tsx.
2. If I then go to that asset in Unity there are two things that can happen. Either it complains that it cannot find the PNG file or it doesn't complain at all. If it doesn't complain I can go back to the map I wanted to import and try again, either it will complain about some other tsx file or it goes through. If however the PNG file wasn't found when the tsx file failed to import, I just click that png file once in Unity and then go back to the tsx file and try again(it always works).

So basically, if it complains about a file, I just click it once in Unity and then it works. It's not ideal but at least I can proceed until there is a fix :)

**Comment by Seanba** [comment]
Hi there, @erikedling. Were you upgrading to Unity 2022 as well?

I've determined that the problem boils down to Unity's built-in texture importer. I may need to reach out to them for a solution.

**Comment by erikedling** [comment]
@Seanba Yes, I upgraded as well

**Comment by Seanba** [comment]
Version 1.10.6 of SuperTiled2Unity fixes this issue. You may need to reimport all your Tiled files, however.
https://seanba.itch.io/supertiled2unity/devlog/411351/supertiled2unity-version-1106

FWIW the problem was SuperTiled2Unity was changing the working directory temporarily while importing Tiled files. This worked "okay" with previous versions of Unity but with Unity 2022 it was causing texture assets to not be imported correctly. It should all be good now.

Thanks for the bug report.
